HeterogeneousCore/AlpakaInterface/interface/prefixScan.h

```diff
@@ -0,0 +1,223 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_prefixScan_h
+#define HeterogeneousCore_AlpakaInterface_interface_prefixScan_h
+// Active version
+
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+namespace cms {
+  namespace alpakatools {
+
+    template <typename TAcc, typename T, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void warpPrefixScan(const TAcc& acc, int32_t laneId, T const* ci, T* co, uint32_t i) {
```

Comment:
The CUDA version computes the lane id internally; was there a particular reason to pass it as a parameter instead ?

Reply:
The warp size goes from a compile time value (hardcoded value 32) to a computed value based on  a variable, so I minimized recomputations.

Reply:
ah, I understand now, thanks !